cpp/src/gandiva/compiled_expr.h

```diff
@@ -18,7 +18,7 @@
 #ifndef GANDIVA_COMPILED_EXPR_H
 #define GANDIVA_COMPILED_EXPR_H
 
-#include <llvm/IR/IRBuilder.h>
+#include "gandiva/llvm_includes.h"
```

Comment:
For the record, when we're only using pointers to LLVM types, we should be able to replace the LLVM includes with forward declarations so as to decrease build times, e.g.:
```cpp
namespace llvm {
  class Function;
}
```

Reply:
Agreed. This was sort of a band-aid solution because of the warning suppression issues. It would be useful to have a file containing LLVM forward declarations that we can use